lib/benchmark/linear.go

```diff
@@ -0,0 +1,48 @@
+package benchmark
+
+import (
+	"errors"
+	"time"
+)
+
+// Linear generator
+type Linear struct {
+	// LowerBound is the lower end of rps to execute
+	LowerBound int
+	// UpperBound is the upper end of rps to execute
+	UpperBound int
+	// Step is the amount of rps to increment by
+	Step int
+	// MinimumMeasurements is the minimum measurement a benchmark should execute
+	MinimumMeasurements int
+	// MinimumWindow is the minimum duration to run benchmark for
+	MinimumWindow time.Duration
+	// Threads is amount of concurrent execution threads to run
+	Threads    int
+	currentRPS int
+	config     Config
+}
+
+// GetBenchmark returns the benchmark config for the current generation.
+func (lg *Linear) GetBenchmark() (Config, error) {
```

Comment:
this method uses an error to signal end of generations, which is not really an error
I suggest changing the return values to `*Config` and returning a nil config at the end.